refactor(minifier): use `map` and `and_then` instead of let else

7086cmd opened the pull request:
For the test case, Closure Compiler doesn't handle this at all in the REPL! If it's necessary, I will turn it back.

This PR uses builtin `and_then` and `map` method, which is better instead of a lot of `if let Some`.

7086cmd commented:
* **#7178** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7178?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @7086cmd and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

camc314 commented on crates/oxc_minifier/src/ast_passes/peephole_replace_known_methods.rs:
```diff
@@ -549,7 +533,8 @@ mod test {
         fold_same("x = 'abcde'.charCodeAt(-1)"); // or x = (0/0)
         fold_same("x = 'abcde'.charCodeAt(y)");
-        fold("x = 'abcde'.charCodeAt()", "x = 97");
+        // Seems that it does not handle this case
+        // fold("x = 'abcde'.charCodeAt()", "x = 97");
```
yeah i tried and_then, but it means you can't handle cases such as this

7086cmd replied:
Yeah. Actually closure compiler doesn't handle this right now in the deprecated REPL. I'll take an investigation on the weekend to determine which approach should we use.

7086cmd replied:
Actually it does not handle it:
```
(base) ... ~ % google-closure-compiler
The compiler is waiting for input via stdin.
x = 'abcde'.charCodeAt()
x="abcde".charCodeAt();
```

Boshen commented on crates/oxc_minifier/src/ast_passes/peephole_fold_constants.rs:
```diff
@@ -63,7 +63,7 @@ impl<'a> Traverse<'a> for PeepholeFoldConstants {
         } {
             *expr = folded_expr;
             self.changed = true;
-        };
```
irrelevant changes?

7086cmd replied:
It might be. I'll fix them later today.

Boshen commented:
### Merge activity

* **Nov 9, 6:47 AM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

7086cmd committed:
refactor(minifier): use `map` and `and_then` instead of let else (#7178)

For the test case, Closure Compiler doesn't handle this at all in the REPL! If it's necessary, I will turn it back.

This PR uses builtin `and_then` and `map` method, which is better instead of a lot of `if let Some`.
